x-pack/elastic-agent/pkg/agent/cmd/container.go

```diff
@@ -541,31 +550,46 @@ func kibanaClient(cfg kibanaConfig, headers map[string]string) (*kibana.Client,
 	}, 0, "Elastic-Agent")
 }
 
-func findPolicy(cfg setupConfig, policies []kibanaPolicy) (*kibanaPolicy, error) {
+func findPolicy(cfg setupConfig, policies []kibanaPolicy, streams *cli.IOStreams) (*kibanaPolicy, error) {
 	policyID := ""
 	policyName := cfg.Fleet.TokenPolicyName
 	if cfg.FleetServer.Enable {
 		policyID = cfg.FleetServer.PolicyID
 	}
+
 	for _, policy := range policies {
 		if policyID != "" {
 			if policyID == policy.ID {
+				logInfo(streams, "using policy with id", policyID)
 				return &policy, nil
 			}
 		} else if policyName != "" {
 			if policyName == policy.Name {
+				logInfo(streams, "using policy named", policyName)
 				return &policy, nil
 			}
 		} else if cfg.FleetServer.Enable {
-			if policy.IsDefaultFleetServer {
+			if policy.ID == cfg.FleetServer.DefaultPolicyID {
+				logWarn(streams, "using default fleet policy id:", policy.ID)
 				return &policy, nil
 			}
 		} else {
-			if policy.IsDefault {
+			if policy.Name == cfg.Fleet.DefaultTokenPolicyName {
+				logWarn(streams, "using default policy name:", policy.Name)
 				return &policy, nil
 			}
 		}
 	}
+
+	if cfg.FleetServer.Enable {
+		if policyID == "" {
+			logWarn(streams, "No policy id set, please set the FLEET_SERVER_POLICY_ID environment variable to the fleet server policy id.")
```

Comment:
here's what i understand this code to mean:

this would be hit if the user didn't configure either a policy id or default when hosting a fleet server under the agent. this gives more specific information to the user assuming the agent fails to find a policy to get a token from.
this coincides with a path that halts start up of the agent.

i'll add comments for both of this situation, and the one below.